addons/l10n_sa/views/report_invoice.xml

```diff
@@ -20,11 +20,17 @@
         </xpath>
         <xpath expr="//t[@t-set='address']" position="after">
             <t t-set="information_block">
-                <p>
-                    <img t-if="o.l10n_sa_qr_code_str"
-                         style="display:block;"
-                         t-att-src="'/report/barcode/?barcode_type=%s&amp;value=%s&amp;width=%s&amp;height=%s'%('QR', quote_plus(o.l10n_sa_qr_code_str), 200, 200)"/>
-                </p>
+                <div class="row">
+                    <p class="col-6 me-3">
+                        <img t-if="o.l10n_sa_qr_code_str"
+                            style="display:block;"
+                            t-att-src="'/report/barcode/?barcode_type=%s&amp;value=%s&amp;width=%s&amp;height=%s'%('QR', quote_plus(o.l10n_sa_qr_code_str), 200, 200)"/>
+                    </p>
+                    <div class="col-6" t-if="o.partner_shipping_id and (o.partner_shipping_id != o.partner_id)" groups="account.group_delivery_invoice_address" name="shipping_address_block">
+                        <strong>Shipping Address:</strong>
+                        <div t-field="o.partner_shipping_id" t-options='{"widget": "contact", "fields": ["address", "name"], "no_marker": True}'/>
+                    </div>
+                </div>
```

Comment:
@yosa-odoo There is no way to extend information_block instead of completely overwriting it?  (Did you consider?)

Reply:
Yes you can but you won't have the same result visually (see my comment on my last review)

Reply:
I agree with @yosa-odoo, visually it won't look nice.